frontend/components/RewardsHistory/useRewardsHistory.ts

```diff
@@ -0,0 +1,124 @@
+import { useQuery } from '@tanstack/react-query';
+import { ethers } from 'ethers';
+import { gql, request } from 'graphql-request';
+import { groupBy } from 'lodash';
+import { z } from 'zod';
+
+import { Chain } from '@/client';
+import { SERVICE_STAKING_TOKEN_MECH_USAGE_CONTRACT_ADDRESSES } from '@/constants/contractAddresses';
+import { STAKING_PROGRAM_META } from '@/constants/stakingProgramMeta';
+import { StakingProgramId } from '@/enums/StakingProgram';
+import { useServices } from '@/hooks/useServices';
+
+import { EpochDetails, StakingRewardSchema } from './types';
+
+const RewardHistoryResponseSchema = z.object({
+  epoch: z.string(),
+  rewards: z.array(z.string()),
+  serviceIds: z.array(z.string()),
+  blockTimestamp: z.string(),
+  transactionHash: z.string(),
+  epochLength: z.number(),
+});
+
+type RewardHistoryResponse = z.infer<typeof RewardHistoryResponseSchema>;
+
+const betaAddress =
+  SERVICE_STAKING_TOKEN_MECH_USAGE_CONTRACT_ADDRESSES[Chain.GNOSIS].pearl_beta;
+const beta2Address =
+  SERVICE_STAKING_TOKEN_MECH_USAGE_CONTRACT_ADDRESSES[Chain.GNOSIS]
+    .pearl_beta_2;
+
+const SUBGRAPH_URL =
+  'https://api.studio.thegraph.com/query/81855/pearl-staking-rewards-history/version/latest';
```

Comment:
a bit concerned with the "latest", in case you'd like to update your subgraph, while it's reindexing the data will be empty. on the other hand, if we specify the version and then update the subgraph a few times, a third from last version will become unavailable. I suggest, in case you want to update the subgraph, just create a new one 🥲

Reply:
you'll have a completely new endpoint once you publish your subgraph